src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -720,6 +728,46 @@ func (r *ClusterReconciler) reconcileClusterForPods(pod client.Object) []reconci
 	}
 }
 
+func (r *ClusterReconciler) reconcileClusterForExternalCASecret(s client.Object) []reconcile.Request {
+	hasExternalCA, found := s.GetAnnotations()[SecretAnnotationExternalCAKey]
+	if !found || hasExternalCA != "true" {
+		return nil
+	}
+
+	clusterName, found := s.GetLabels()[labels.InstanceKey]
+	if !found {
+		return nil
+	}
```

Comment:
If there is cluster name in annotation I don't see any need to list all clusters and find the one from annotation.

Reply:
The purpose of the listing is to find that the cluster exists, so that the controller does not receive request for a cluster that does not exist.  We could add an annotation that specifies the namespace, then we would call Get() instead of List().

Reply:
I was under impression that https://github.com/redpanda-data/redpanda/pull/9863#discussion_r1160278469 and other comments around controller being namespace constraint will be addressed in this PR. Then namespace will be the same as the secret resource, right?

Reply:
You are right. I guess my mind was still thinking about that the secret can be in different namespace. Good catch!
I will update in the next commit.

Reply:
Updated.